keps/sig-node/3619-supplemental-groups-policy/README.md

```diff
@@ -790,13 +790,44 @@ rollout. Similarly, consider large clusters and how enablement/disablement
 will rollout across nodes.
 -->
 
+A rollout may fail when at least one of the following components are too old because this KEP introduces the new Kubernetes API field:
+
+| Component      | `supplementalGroupsPolicy` value that will cause an error |
+|----------------|-----------------------------------------------------------|
+| kube-apiserver | not null                                                  |
+| kubelet        | not null                                                  |
+| CRI runtime    | `Strict`                                                  |
+
+
+For example, an error will be returned like this if kube-apiserver is too old:
+```console
+$ kubectl apply -f supplementalgroupspolicy.yaml
+Error from server (BadRequest): error when creating "supplementalgroupspolicy.yaml": Pod in version "v1" cannot be handled as a Pod:
+strict decoding error: unknown field "spec.securityContext.supplementalGroupsPolicy"
+```
+
+No impact on already running workloads.
+
 ###### What specific metrics should inform a rollback?
 
 <!--
 What signals should users be paying attention to when the feature is young
 that might indicate a serious problem?
 -->
 
+Look for an event saying indicating SupplementalGroupsPolicy is not supported by the runtime.
+```console
+$ kubectl get events -o json -w
+...
+{
+    ...
+    "kind": "Event",
+    "message": "Error: SupplementalGroupsPolicyNotSupported",
+    ...
+}
+...
+```
```

Comment:
I plan to add kubelet admission which raises an error event if
- the pod sets `pod.spec.securityContext.supplementalGroupsPolicy`
- and the node is `node.status.features.supplementalGroupsPolicy=false`

Reply:
ok, and so the pod doesn't get admitted, can there be metric when this happens for this reason? Events are not useful if you are managing 50,000 clusters...

Reply:
> can there be metric when this happens for this reason?

hmm. If I understood correctly, there are currently no metrics for specific kubelet's admission errors. This kind of admission error is a common issue for node(or runtime handler) feature-based admission (e.g. user namespace or recursive read-only mounts).

Like [KEP-127: Support User Namespaces](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/127-user-namespaces/README.md#what-specific-metrics-should-inform-a-rollback) stated, we can compare `kubelet_running_pods` and `kubelet_desired_pods` metrics.

WDYT??

Reply:
yeah, that looks reasonable, please pull that in here. thanks